wg-api/spec-documents/clipboard-rearchitecture.md

```diff
@@ -0,0 +1,285 @@
+3. Leave API as-is and continue to adapt it as required by upstream changes by Chromium development.
+ 
+### Preferred Approach/Solution
+
+Alignment with W3C where possible, with some additional APIs exposed to handle desktop use cases which aren’t considered by the specification.
+
+**APIs to Preserve**
+* `clipboard.clear([type])`
+  * Not supported through existing Web APIs or in the specification.
+* `clipboard.readText([type])`
+  * Already implemented per specification.
+* `clipboard.writeText(text[, type])`
+  * Already implemented per specification.
+
+**APIs to Remove**
+* `clipboard.availableFormats([type])`
+  * Superseded by `clipboard.read` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.has(format[, type])` 
+  * Superseded by `clipboard.read` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.readBookmark()`
+  * Superseded by `clipboard.read` with custom `electron/bookmark` MIME type
+* `clipboard.writeBookmark(title, url[, type])`
+  * Superseded by `clipboard.write` with custom `electron/bookmark` MIME type
+* `clipboard.readBuffer(format)`
+  * Superseded by `clipboard.read`[Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read)
+* `clipboard.writeBuffer(format, buffer[, type])`
+  * Superseded by `clipboard.write` [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/write)
+* `clipboard.readFindText()` _macOS_
+  * Superseded by `clipboard.read` with custom `electron/findtext` MIME type
+* `clipboard.writeFindText(text)`
+  * Superseded by `clipboard.write` with custom `electron/findtext` MIME type
+* `clipboard.readHTML()`
+  * Superseded by `clipboard.read` with `text/html` MIME type
+* `clipboard.writeHTML(markup[, type])`
+  * Superseded by `clipboard.write` with `text/html` MIME type
+* `clipboard.readImage([type])`
+  * Superseded by `clipboard.read()` with `image/*` (where * is bmp, gif, jpeg, png, svg+xml, etc.)
+* `clipboard.writeImage(image[, type])`
+  * Superseded by `clipboard.write`  with `image/*` (where * is bmp, gif, jpeg, png, svg+xml, etc.)
+* `clipboard.readRTF([type])`
+  * Superseded by `clipboard.read` with `application/rtf` MIME type
+* `clipboard.writeRTF(text[, type])`
+  * Superseded by `clipboard.write` with `application/rtf` MIME type
+
+**APIs to Modify**
+* `clipboard.read([clipboardType])`
+  * Modify to bring into spec compliance with [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read) `clipboard.read`
+  * Ensure that raw formats are preserved
+    * Custom MIME types e.g. `electron/filepath`
+  * clipboardType is only used for Linux to specify if clipboard is regular clipboard or `selection` clipboard.
+* `clipboard.write(data[, clipboardType]])​`
+  * Modify to bring into spec compliance with [Web API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read) `clipboard.read`
+  * Ensure that raw formats are preserved
```

Comment:
Burndown output - we shouldn't use mystery mime types for types which represent location as opposed to content type. We should instead add Electron-specific clipboard-ish functionality as discrete methods alongside coalescing existing MIME types into methods aligned with the Web APIs.

Reply:
Using custom mime types corresponds to what the web is doing via custom clipboard 
 types: https://github.com/w3c/editing/blob/gh-pages/docs/clipboard-pickling/explainer.md

Reply:
cc @nornagon iirc you strongly preferred avoiding custom mime types - what are your thoughts given the above? I'd also like to choose vaguely the same approach as upstream, at minimum for future-proofing purposes

Reply:
Honestly I don't remember what my objection was; following the Web standard seems like a reasonable thing to do. Notably, the Web standard looks like it's only allowing _very circumscribed_ custom formats, in the pattern `web foo/bar`. Which I don't think was what this proposal was outlining.